packages/remix-dev/config.ts

```diff
@@ -128,6 +128,13 @@ export interface AppConfig {
    * routes.
    */
   ignoredRouteFiles?: string[];
+
+  /**
+   * A list of module names or regular expressions that determined if a module
+   * is transpiled and included in the server bundle. This can be useful when
+   * consuming ESM only packages in a CJS build.
+   */
+  transpileModules?: Array<string | RegExp>;
```

Comment:
Is there a different name to give this property to make it more clear that:

1. This only applies to the server build
2. This only applies to node_modules

Reply:
I'm down to call it "transpileServerBareModules" or similar, but the thing is that you're never(?) going to be given this option for a client build as that all needs to be bundled regardless.

Reply:
How about `serverDependenciesToBundle`?

Things I like about it:

- it starts with `server`, just like all our other server-only config vars
- it specifically talks about dependencies (node_modules)

Reply:
What about `serverDependenciesToTranspile`? I like the "transpile" over "bundle" for a few reasons:
- it's more than just bundling, it does in-fact transpile ESM->CJS
- prior IP refers to bundling / transforming dependencies as transpile and might be easier for people to find / use https://www.npmjs.com/package/next-transpile-modules

Reply:
I prefer `serverDependenciesToBundle`:

1. It's doing more than "transpiling"
2. "Bundle" suggests also compiling in some way
3. "Transpile" is a made-up word that I never liked 😆

Reply:
In any case though, I wouldn't stop the ship over this. I'm fine with either.